core/src/test/scala/org/apache/spark/deploy/StandaloneDynamicAllocationSuite.scala

```diff
@@ -395,8 +395,8 @@ class StandaloneDynamicAllocationSuite
     }
 
     var apps = getApplications()
-    // kill executor 1
-    assert(sc.killExecutor(executors.head))
+    // kill executor 1, and actually nothing to kill
+    assert(!sc.killExecutor(executors.head))
```

Comment:
no, the idea is more like the following:

- you start with executors {27, 28}
- you kill and replace 27, so you end up with executors {28, 29}
- now you want to kill 28, this should succeed (but currently it doesn't in the tests)